expression/builtin_other.go

```diff
@@ -78,10 +78,11 @@ func (b *builtinInSig) eval(row []types.Datum) (d types.Datum, err error) {
 			hasNull = true
 			continue
 		}
-
 		a, b, err := types.CoerceDatum(sc, args[0], v)
 		if err != nil {
-			return d, errors.Trace(err)
+			// If the coerce fail, we can't raise error.
```

Comment:
`CoerceDatum` should deal with `truncated error` internally, and return nil as error.
But here it returns `BadNumber` when coerce fail. So the real problem is in `CoerceDatum`, not here. I will keep old code here, and fix the problem in `CoerceDatum`.